python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -114,15 +114,21 @@ def _attribute_filter__set__(self, name, value):
             #   Do some checking of the filenames in a subprocess
             if name == 'lfn':
                 self.namePattern = os.path.basename(value)
-                self.remoteDir = os.path.dirname(value)
+                if os.path.dirname(value):
+                    self.remoteDir = os.path.dirname(value)
+                elif os.path.isfile(os.path.join(os.getcwd(), self.namePattern)):
+                    self.remoteDir = os.getcwd()
                 return value
 
             elif name == 'namePattern':
                 self.localDir = os.path.dirname(value)
                 return os.path.basename(value)
 
             elif name == 'localDir':
-                return expandfilename(value)
+                if value:
+                    return expandfilename(value)
+                else:
+                    return value
```

Comment:
No because then we can never set `''` as a directory either during construction/tests as a user as this will always be auto-expanded to use the cwd.